### ISSUE TITLE
Buy on PACER doesn't always give a RECAP upload

### ISSUE THREAD

**Issue by troglodite2** [freelawproject/recap#336]
Some "Buy on PACER" buttons take you to PACER pages that do not have a download link.

See https://github.com/freelawproject/recap/issues/337 for details on why this is an issue.

This https://ecf.txwd.uscourts.gov/doc1/181030035509?caseid=1016479 was attached to a "Buy on Pacer" button on https://www.courtlistener.com/docket/14793743/cargill-v-barr/?order_by=desc Document Number 73.  

While logged into PACER, this took me to a PACER warning that I was using an outside link that could result in a charge.

Once on the PACER page, I was on the charge page telling me how much it would cost with only a "view selected" option.  No RECAP upload.

**Comment by mlissner** [comment]
When I press the "View Selected" option, it uploaded. That didn't do it for you?

**Comment by ERosendo** [comment]
@mlissner I talked to @troglodite2 in slack and explained to him that the extension should be able to upload documents if someone uses the "view document" in this case because that button will give us a single pdf ( not a combined pdf like in the other issue).

He's still not able to upload documents with the extension, so I asked him to provide a list of extensions he's installed in his browser and the version of firefox. Here's the list of extension:

- LanguageTool, not active on pacer.  
- Keeper, active on Pacer but only processes <input type="password"...> 
- Markdown Here, only active on Google Mail, 
- Recap, only active on PACER sites, 
- Xdebug Helper, not active on PACER or CL.  
- Grammarly, manually disabled.

and he's using Firefox Version 111.01 20230321162322. I will try to replicate the issue.

**Comment by mlissner** [comment]
Cool. @troglodite2, can you get a fresh FF profile, install RECAP, and see if you can still reproduce? If so, we need to do a binary search to figure out what the problem is. 

There is a list of known-bad extensions here: https://free.law/recap/faq, but I sort of don't think that's it given that most of yours are limited in scope.

**Comment by troglodite2** [comment]
I did one better and installed Ubuntu 22.10 on a VM from scratch, then ran FF for the first time.  Logged into the CL site and installed RECAP Add-on.  I then went to a case and clicked the "Buy on Pacer" logged in, clicked on the "view selected" and it just worked.  Complete with the floating notification that @ERosendo said I should see.

I'm going to install the extensions I don't think will cause and issue and test again.

**Comment by troglodite2** [comment]
Keeper and markdown added.  No change.  It still worked.
Added LaungugeTool and x-debug helper.  Still works.

Important difference.  I'm seeing that the Recap add-on is blue all the time on the pacer site.

**Comment by troglodite2** [comment]
It looks like the recap extension failed to detect that it was on a PACER page.  When this happens, downloading fails to upload.

My fix was to quit out of Firefox.  Perform a `snap refresh` as root.  Then restart Firefox.  

My machine has 32 GB of ram, but I am a developer with way too many tabs open.  I am thinking that the Recap Extension just did not get enough cycles to work.  I'm not sure how to reproduce the issue.

My current workaround is to verify that I have a blue recap extension before clicking anything on PACER.

**Comment by mlissner** [comment]
Well, that's weird, but OK. It doesn't take many cycles for RECAP to operate, but if you're out of memory, the background page where RECAP runs could crash I guess. 

If you can't reproduce, shall we close this?

**Comment by troglodite2** [comment]
I do not think it is safe to close this yet.  This is the second time it happened to me.  I did not have any OOM events.  Just things being slower than I want.  I believe that there is something that happens when recap add-on can't detect that it is on a PACER page.  I want to investigate that further when it happens again.
